src/components/Layout/GridTableLayout/GridTableLayout.tsx

```diff
@@ -89,14 +99,43 @@ function GridTableLayoutComponent<
   X extends Only<GridTableXss, X>,
   QData,
 >(props: GridTableLayoutProps<F, R, X, QData>) {
-  const { pageTitle, breadcrumb, tableProps, layoutState, primaryAction, secondaryAction, tertiaryAction } = props;
+  const {
+    pageTitle,
+    breadcrumb,
+    tableProps,
+    layoutState,
+    primaryAction,
+    secondaryAction,
+    tertiaryAction,
+    hideEditColumns = false,
+  } = props;
 
+  const tid = useTestIds(props);
+  const columns = tableProps.columns;
+
+  !hideEditColumns && validateColumns(columns);
+
+  const hasHideableColumns = useMemo(() => columns.some((c) => c.canHide), [columns]);
+
+  // Use user-provided API if available, otherwise create our own
+  const defaultApi = useGridTableApi<R>();
+  const api = tableProps.api ?? defaultApi;
```

Comment:
Given we're in beam internals, we could probably do what GridTable does:

```
  const api = useMemo<GridTableApiImpl<R>>(
    () => {
      // Let the user pass in their own api handle, otherwise make our own
      const api = (props.api as GridTableApiImpl<R>) ?? new GridTableApiImpl();
  }, [api]);
```

Instead of calling `useGridTableApi()`